File: Level2/16.큰수만들기/other.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k)
{
    string answer = "";
    answer = number.substr(k);

    for (int i = k - 1; i >= 0; i--)
    {
        int j = 0;
        while (1)
        {
            if (number[i] >= answer[j])
            {
                char temp = number[i];
                number[i] = answer[j];
                answer[j] = temp;
                j++;
            }
            else
                break;
        }
    }

    return answer;
}

File: Level3/2.N으로표현/other.js
function solution(N, number) {
  const cache = Array.from({ length: 9 }, () => new Set());

  for (let i = 1; i <= 8; i++) {
    cache[i].add('1'.repeat(i) * N);

    for (let j = 1; j < i; j++) {
      for (const arg1 of cache[j]) {
        for (const arg2 of cache[i - j]) {
          cache[i].add(arg1 + arg2);
          cache[i].add(arg1 - arg2);
          cache[i].add(arg1 * arg2);
          cache[i].add(parseInt(arg1 / arg2));
        }
      }
    }

    if (cache[i].has(number)) return i;
  }

  return -1;
}
